Modules/Filtering/Convolution/test/itkConvolutionImageFilterStreamingTest.cxx

```diff
@@ -0,0 +1,173 @@
+  auto           source = SourceType::New();
+  KernelSizeType kernelSize{ 3, 5 };
+  source->SetSize(kernelSize);
+  source->SetMean(2);
+  source->SetSigma(3.0);
+  source->SetScale(1.0);
+  source->SetNormalized(true);
+  source->Update();
+  return source->GetOutput();
+}
+
+template <typename ConvolutionFilterType>
+int
+doConvolutionImageFilterStreamingTest(int argc, char * argv[])
+{
+  constexpr int ImageDimension = 2;
+
+  using PixelType = float;
+  using ImageType = itk::Image<PixelType, ImageDimension>;
+  using RegionType = typename ImageType::RegionType;
+  using SizeType = typename RegionType::SizeType;
+  using IndexType = typename RegionType::IndexType;
+
+  // Request a subregion of the largest possible output
+  IndexType requestedIndex;
+  requestedIndex[0] = std::atoi(argv[4]);
+  requestedIndex[1] = std::atoi(argv[5]);
+  SizeType requestedSize;
+  requestedSize[0] = std::atoi(argv[6]);
+  requestedSize[1] = std::atoi(argv[7]);
+  RegionType requestedRegion(requestedIndex, requestedSize);
+
+  using ReaderType = itk::ImageFileReader<ImageType>;
+
+  auto reader1 = ReaderType::New();
+  reader1->SetFileName(argv[2]);
+
+  // Verify the requested region is a subregion of the largest possible region
+  reader1->UpdateOutputInformation();
+  ITK_TEST_EXPECT_TRUE(reader1->GetOutput()->GetLargestPossibleRegion().IsInside(requestedRegion));
+
+  using PipelineMonitorType = itk::PipelineMonitorImageFilter<ImageType>;
+  auto inputMonitor = PipelineMonitorType::New();
+  inputMonitor->SetInput(reader1->GetOutput());
+
+  KernelImageType::Pointer kernelImage = GenerateKernelForStreamingTest();
+
+  auto convoluter = ConvolutionFilterType::New();
+  convoluter->SetInput(inputMonitor->GetOutput());
+  convoluter->SetKernelImage(kernelImage);
+  convoluter->SetReleaseDataFlag(true);
+  itk::SimpleFilterWatcher watcher(convoluter, "filter");
+
+  using PipelineMonitorType = itk::PipelineMonitorImageFilter<ImageType>;
+  auto outputMonitor = PipelineMonitorType::New();
+  outputMonitor->SetInput(convoluter->GetOutput());
+
+  using StreamingFilterType = itk::StreamingImageFilter<ImageType, ImageType>;
+  auto streamer = StreamingFilterType::New();
+  streamer->SetInput(outputMonitor->GetOutput());
+  const unsigned int numStreamDivisions = 10;
+  streamer->SetNumberOfStreamDivisions(numStreamDivisions);
+  streamer->GetOutput()->SetRequestedRegion(requestedRegion);
+  ITK_TRY_EXPECT_NO_EXCEPTION(streamer->Update());
+
+  // Verify the pipeline executed as expected
+  const unsigned int numUpdates = 1;
+  // Expect requested region propagates twice due to ConvolutionImageFilter having
+  // two inputs (Primary and KernelImage)
+  const unsigned int numOutputRequestedRegions = 2;
+  // Verify ConvolutionImageFilter upstream requests
+  ITK_TEST_EXPECT_EQUAL(inputMonitor->GetOutputRequestedRegions().size(), numOutputRequestedRegions);
+  ITK_TEST_EXPECT_EQUAL(inputMonitor->GetNumberOfUpdates(), numUpdates);
```

Comment:
Don't we expect that the Convolution filter makes 10 different requested regions upstream? Causing the reader to read 10 regions. This looks like it just made one request and the inputs/reader read the whole image. Or maybe the reader can only read the whole image?

The reader's behavior is dependent on the type of input image, if the underlying ImageIO supports streaming. MetaIO should, as long as the image in not compressed.

Reply:
Yes, good catch. I dug in and addressed two issues:
1. Tests were using `cthead1.png` and it appears streaming was not supported for PNG format. I have added `cthead1.mha` to ITKTestingData for this test, streaming appears to execute properly now.
2. After running into issues with the streamed requested region apparently exceeding the largest possible input region, I was able to track down where `itk::ConvolutionImageFilter` previously did not set the correct output region. Updated accordingly.